### ISSUE TITLE
Crash after serverstop/serverstart while MusicBot is on the server

### ISSUE THREAD

**Issue by lexesv** [TeaSpeak/TeaSpeak#56]
## System Information
***Operating System***: debian 9
***Operating System Type*** (x86|x64): x64
***TeaSpeak Version***: 1.0.29-beta-3

## Issue
***crash after serverstop/serverstart while MusicBot is on the server***:

```
  Error Info:
    Error type        : Signal
    Number            : 0
    Message           : Success
    Code              : 1

  Backtrace:
[16:33:03] [ERROR]   1 | Found delay while ticking server. (Delay: 14671ms)
[16:33:03] [INFO]     | [WebList] Sending new report.
[ 1]    [./TeaSpeakServer                     ] [0x000000000073fece] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 2]    [./TeaSpeakServer                     ] [0x00000000004d7bc4] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 3]    [./TeaSpeakServer                     ] [0x00000000004d437c] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 4]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007f86c14840c0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x110c0) [0x7f86c14840c0]
[ 5]    [./TeaSpeakServer                     ] [0x0000000000573f5a] std::__lg(long)                                                                                                                                                                                                                                                                                     at ??:0
[ 6]    [./TeaSpeakServer                     ] [0x00000000006ec88d] std::_Sp_counted_ptr<YAML::detail::memory*, (__gnu_cxx::_Lock_policy)2>::_M_get_deleter(std::type_info const&)                                                                                                                                                                                      at ??:0
[ 7]    [./TeaSpeakServer                     ] [0x000000000070fa26] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 8]    [./TeaSpeakServer                     ] [0x00000000004b7418] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 9]    [./TeaSpeakServer                     ] [0x000000000065474d] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[10]    [./TeaSpeakServer                     ] [0x000000000064a202] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[11]    [./TeaSpeakServer                     ] [0x0000000000471a0f] char* std::__copy_move_a<false, char*, char*>(char*, char*, char*)                                                                                                                                                                                                                                  at ??:0
[12]    [./TeaSpeakServer                     ] [0x0000000000645917] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[13]    [./TeaSpeakServer                     ] [0x000000000064538d] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[14]    [./TeaSpeakServer                     ] [0x0000000000642df4] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[15]    [./TeaSpeakServer                     ] [0x0000000000642e24] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[16]    [./TeaSpeakServer                     ] [0x00000000007bc1dc] std::experimental::filesystem::v1::path::_M_split_cmpts()                                                                                                                                                                                                                                           at ??:0
[17]    [./TeaSpeakServer                     ] [0x0000000000631260] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[18]    [./TeaSpeakServer                     ] [0x0000000000636ede] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[19]    [./TeaSpeakServer                     ] [0x0000000000636e12] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[20]    [./TeaSpeakServer                     ] [0x0000000000636da2] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[21]    [./TeaSpeakServer                     ] [0x0000000000739827] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[22]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007f86c147a494] /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7f86c147a494]
[23]    [/lib/x86_64-linux-gnu/libc.so.6      ] [0x00007f86c0545aff] /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f86c0545aff]
---------------[ Crash log ]---------------
[16:33:03] [ERROR]     | [WebList] Invalid weblist status response length (-1) Error Resource temporarily unavailable

```

**Comment by WolverinDEV** [comment]
I fill fix this crash later :)
Im still working on my web client currently
![](http://img.mcgalaxy.de/Screenshot-17-50-37.png)

**Comment by lexesv** [comment]
Cool!
Well, I'm just reporting crashes. We discussed this in.

**Comment by WolverinDEV** [comment]
I know,
and im happy about it that you're so dedicated :+1:

**Comment by lexesv** [comment]
...crash even if bot does not play

**Comment by lexesv** [comment]
**v1.1.0-beta-6**

```
---------------[ Crash log ]---------------
  General Info:
     Version           : 1.1.0-beta
     Current Timestamp : 24-02-2018 11:32:44
     System version    : sn: Linux r: 4.9.0-5-amd64 v: #1 SMP Debian 4.9.65-3+deb9u2 (2018-01-04) m: x86_64 d: (none)

  Error Info:
    Error type        : Signal
    Number            : 0
    Message           : Success
    Code              : 1

  Backtrace:
[ 1]    [./TeaSpeakServer                     ] [0x000000000076fb5c] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 2]    [./TeaSpeakServer                     ] [0x00000000004dc594] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 3]    [./TeaSpeakServer                     ] [0x00000000004d8d4c] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 4]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007f1b48f830c0] /lib/x86_64-linux-gnu/libpthread.so.0(+0x110c0) [0x7f1b48f830c0]
[ 5]    [./TeaSpeakServer                     ] [0x0000000000578876] std::__lg(long)                                                                                                                                                                                                                                                                                     at ??:0
[ 6]    [./TeaSpeakServer                     ] [0x00000000006f36b7] std::_Sp_counted_ptr<YAML::detail::memory*, (__gnu_cxx::_Lock_policy)2>::_M_get_deleter(std::type_info const&)                                                                                                                                                                                      at ??:0
[ 7]    [./TeaSpeakServer                     ] [0x00000000007167f4] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[ 8]    [./TeaSpeakServer                     ] [0x00000000004bbaa3] std::basic_string<char, std::char_traits<char>, std::allocator<char> > std::operator+<char, std::char_traits<char>, std::allocator<char> >(std::basic_string<char, std::char_traits<char>, std::allocator<char> >&&, std::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) at ??:0
[ 9]    [./TeaSpeakServer                     ] [0x000000000065938b] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[10]    [./TeaSpeakServer                     ] [0x000000000064ee40] void spdlog::logger::log<std::string>(spdlog::level::level_enum, std::string const&)                                                                                                                                                                                                                at ??:0
[11]    [./TeaSpeakServer                     ] [0x000000000047534b] char* std::__copy_move_a<false, char*, char*>(char*, char*, char*)                                                                                                                                                                                                                                  at ??:0
[12]    [./TeaSpeakServer                     ] [0x000000000064a555] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[13]    [./TeaSpeakServer                     ] [0x0000000000649fcb] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[14]    [./TeaSpeakServer                     ] [0x0000000000647a32] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[15]    [./TeaSpeakServer                     ] [0x0000000000647a62] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[16]    [./TeaSpeakServer                     ] [0x00000000007ecbc4] std::experimental::filesystem::v1::path::_M_split_cmpts()                                                                                                                                                                                                                                           at ??:0
[17]    [./TeaSpeakServer                     ] [0x00000000007ed707] std::experimental::filesystem::v1::path::_M_split_cmpts()                                                                                                                                                                                                                                           at ??:0
[18]    [./TeaSpeakServer                     ] [0x0000000000635e9e] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[19]    [./TeaSpeakServer                     ] [0x000000000063bb1c] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[20]    [./TeaSpeakServer                     ] [0x000000000063ba50] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[21]    [./TeaSpeakServer                     ] [0x000000000063b9e0] bool __gnu_cxx::operator==<char*, std::string>(__gnu_cxx::__normal_iterator<char*, std::string> const&, __gnu_cxx::__normal_iterator<char*, std::string> const&)                                                                                                                                    at ??:0
[22]    [./TeaSpeakServer                     ] [0x000000000076970f] void std::_Destroy<std::shared_ptr<music::manager::PlayerProvider> >(std::shared_ptr<music::manager::PlayerProvider>*)                                                                                                                                                                              at ??:0
[23]    [/lib/x86_64-linux-gnu/libpthread.so.0] [0x00007f1b48f79494] /lib/x86_64-linux-gnu/libpthread.so.0(+0x7494) [0x7f1b48f79494]
[24]    [/lib/x86_64-linux-gnu/libc.so.6      ] [0x00007f1b47d73aff] /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f1b47d73aff]
---------------[ Crash log ]---------------
[11:32:44] [ERROR]     | [WebList] Invalid web client key response (-1) Error Resource temporarily unavailable
```

**Comment by WolverinDEV** [comment]
:)  I will resolve this issue now :)

**Comment by lexesv** [comment]
Solved in TeaSpeak-Server v1.1.0-beta-7
@WolverinDEV Thanks!
